utils/calendrical_calculations/src/chinese_based.rs

```diff
@@ -269,7 +269,18 @@ pub(crate) fn midnight<C: ChineseBased>(moment: Moment) -> Moment {
 pub(crate) fn new_year_in_sui<C: ChineseBased>(prior_solstice: RataDie) -> (RataDie, RataDie) {
     // s1 is prior_solstice
     // Using 370 here since solstices are ~365 days apart
+    // Both solstices should fall in December
```

Comment:
This assertion fails for years before about 20,000 BCE. Is this WAI? I'm not sure which factor influences this more:

1. Although the Gregorian calendar drifts by less than Julian, it still drifts. Do we hit a threshold around 20,000 BCE where the solstice moved into November?
2. Or is this just an artifact of floating point error in astronomical calculations?
3. Or are we getting sidereal drift?

Reply:
This is WAI but we could make the assertion stop failing outside that range, or clamp the Chinese calendar.

AIUI the Gregorian calendar accounts for the precession of the equinoxes (the biggest factor here). There are some other smaller factors that lead to the 3216 number, but the thing is that at that scale the planetary bodies _are not that predictable_. Basically I'm not convinced there's a sensible way to actually make any predictions about dates that far off for something like this.